src/backing/client/grpc_client/proto_util_test.cc

```diff
@@ -0,0 +1,208 @@
+  "an arbitrary digest",
+  "",
+};
+
+class MakeDigestTest : public
+    testing::TestWithParam<std::tuple<DigestCase, std::string>> {
+  // Purposely empty; no fixtures to instantiate.
+};
+
+INSTANTIATE_TEST_SUITE_P(
+    DigestParameters, MakeDigestTest,
+    Combine(Values(DigestCase::kSha256,
+                   DigestCase::kSha384,
+                   DigestCase::kSha512),
+            ValuesIn(kSampleDigests)));
+
+TEST_P(MakeDigestTest, MakeDigest) {
+  auto expected_digest_type = std::get<0>(GetParam());
+  auto expected_digest_bytes = std::get<1>(GetParam());
+
+  auto actual = MakeDigest(expected_digest_type, expected_digest_bytes);
+  EXPECT_EQ(FromProtoToDigestCase(actual.digest_case()), expected_digest_type);
+  EXPECT_THAT(GetDigestBytes(actual), StrEq(expected_digest_bytes));
+}
+
+// Mapping between `StatusCode` cases and their protobuf equivalents.
+struct CorrespondingStatusCode {
+  grpc::StatusCode proto;
+  StatusCode code;
+} kStatusCodeMapping[]{
+    {grpc::StatusCode::OK, StatusCode::kOk},
+    {grpc::StatusCode::CANCELLED, StatusCode::kCancelled},
+    {grpc::StatusCode::UNKNOWN, StatusCode::kUnknown},
+    {grpc::StatusCode::INVALID_ARGUMENT, StatusCode::kInvalidArgument},
+    {grpc::StatusCode::DEADLINE_EXCEEDED, StatusCode::kDeadlineExceeded},
+    {grpc::StatusCode::NOT_FOUND, StatusCode::kNotFound},
+    {grpc::StatusCode::ALREADY_EXISTS, StatusCode::kAlreadyExists},
+    {grpc::StatusCode::PERMISSION_DENIED, StatusCode::kPermissionDenied},
+    {grpc::StatusCode::UNAUTHENTICATED, StatusCode::kUnauthenticated},
+    {grpc::StatusCode::RESOURCE_EXHAUSTED, StatusCode::kResourceExhausted},
+    {grpc::StatusCode::FAILED_PRECONDITION, StatusCode::kFailedPrecondition},
+    {grpc::StatusCode::ABORTED, StatusCode::kAborted},
+    {grpc::StatusCode::OUT_OF_RANGE, StatusCode::kOutOfRange},
+    {grpc::StatusCode::UNIMPLEMENTED, StatusCode::kUnimplemented},
+    {grpc::StatusCode::INTERNAL, StatusCode::kInternal},
+    {grpc::StatusCode::UNAVAILABLE, StatusCode::kUnavailable},
+    {grpc::StatusCode::DATA_LOSS, StatusCode::kDataLoss},
+};
+
+class FromGrpcStatusToStatusTest : public
+    testing::TestWithParam<CorrespondingStatusCode> {
+  // Purposely empty; no fixtures to instantiate.
+};
+
+INSTANTIATE_TEST_SUITE_P(StatusCodeParameters, FromGrpcStatusToStatusTest,
+                         ValuesIn(kStatusCodeMapping));
+
+TEST_P(FromGrpcStatusToStatusTest, ConversionsWork) {
+  auto mapping = GetParam();
+  std::string const message = "test message";
+  auto const original = grpc::Status(mapping.proto, message);
+  auto const expected = Status(mapping.code, message);
+  auto const actual = FromGrpcStatusToStatus(original);
+  EXPECT_EQ(expected, actual);
+}
+
+using ProtoDigestCase = google::cloud::kms::v1::Digest::DigestCase;
+
+// Mapping between `DigestCase` cases and their protobuf equivalents.
+struct CorrespondingDigestCase {
+  DigestCase expected;
+  ProtoDigestCase proto;
+} const kDigestMapping[]{
```

Comment:
Nit: separate type definition from variable declaration as discussed previously

Reply:
Handling in #77.